kernel/cClosure.ml

```diff
@@ -602,7 +602,7 @@ and to_constr_case lfts ci u pms (p,r) iv c ve env =
             Array.map f_ctx ve)
 
 and comp_subs el (s,u') =
-  Esubst.lift_subst (fun el c -> lazy (to_constr el c)) el s, u'
+  Esubst.lift_subst (fun el c -> lazy (Vars.make_substituend @@ to_constr el c)) el s, u'
```

Comment:
We could probably be even more clever here by making to_constr return a substituend so as to share more lifts.